app/src/main/java/org/greenstand/android/TreeTracker/models/messages/MessagesRepo.kt

```diff
@@ -146,24 +152,46 @@ class MessagesRepo(
             limit = query.limit,
         )
         query = result.query
+        if (query.total == 0) return@withContext
+
+        val saveMessageResponseJobs = mutableListOf<Job>()
+
+        result.messages.forEach {
+            ensureActive()
+            // launch all jobs in parallel and add these to a list.
+            saveMessageResponseJobs.add(launch {
+                saveMessageResponse(wallet, it.copy())
+            })
+        }
```

Comment:
I'm not too familiar with using jobs. What is the difference between doing this and using async such as:
messages.map { async { saveMessage() } }.forEach{ it.await() }

Reply:
they are the same thing in this case. await is for when we have to get a result back from a method eg: fun saveMessages(message) : Boolean but here the method doesn't return anything hence we use launch.

Reply:
Let's use async then, it's used for parallel work and removes the need to maintain a list of jobs.